File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/update_ontology_project_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class UpdateOntologyProjectDetails(object):
    """
    UpdateOntologyProjectDetails model.
    """

    #: A constant which can be used with the source_type property of a UpdateOntologyProjectDetails.
    #: This constant has a value of "VOLUME"
    SOURCE_TYPE_VOLUME = "VOLUME"

    #: A constant which can be used with the source_type property of a UpdateOntologyProjectDetails.
    #: This constant has a value of "GIT"
    SOURCE_TYPE_GIT = "GIT"

    def __init__(self, **kwargs):
        """
        Initializes a new UpdateOntologyProjectDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this UpdateOntologyProjectDetails.
        :type key: str

        :param display_name:
            The value to assign to the display_name property of this UpdateOntologyProjectDetails.
        :type display_name: str

        :param description:
            The value to assign to the description property of this UpdateOntologyProjectDetails.
        :type description: str

        :param namespace:
            The value to assign to the namespace property of this UpdateOntologyProjectDetails.
        :type namespace: str

        :param workspace_base_path:
            The value to assign to the workspace_base_path property of this UpdateOntologyProjectDetails.
        :type workspace_base_path: str

        :param source_type:
            The value to assign to the source_type property of this UpdateOntologyProjectDetails.
            Allowed values for this property are: "VOLUME", "GIT"
        :type source_type: str

        :param git_repository_key:
            The value to assign to the git_repository_key property of this UpdateOntologyProjectDetails.
        :type git_repository_key: str

        :param git_branch_name:
            The value to assign to the git_branch_name property of this UpdateOntologyProjectDetails.
        :type git_branch_name: str

        :param git_folder_path:
            The value to assign to the git_folder_path property of this UpdateOntologyProjectDetails.
        :type git_folder_path: str

        :param lifecycle_state:
            The value to assign to the lifecycle_state property of this UpdateOntologyProjectDetails.
        :type lifecycle_state: str

        """
        self.swagger_types = {
            'key': 'str',
            'display_name': 'str',
            'description': 'str',
            'namespace': 'str',
            'workspace_base_path': 'str',
            'source_type': 'str',
            'git_repository_key': 'str',
            'git_branch_name': 'str',
            'git_folder_path': 'str',
            'lifecycle_state': 'str'
        }

        self.attribute_map = {
            'key': 'key',
            'display_name': 'displayName',
            'description': 'description',
            'namespace': 'namespace',
            'workspace_base_path': 'workspaceBasePath',
            'source_type': 'sourceType',
            'git_repository_key': 'gitRepositoryKey',
            'git_branch_name': 'gitBranchName',
            'git_folder_path': 'gitFolderPath',
            'lifecycle_state': 'lifecycleState'
        }

        self._key = None
        self._display_name = None
        self._description = None
        self._namespace = None
        self._workspace_base_path = None
        self._source_type = None
        self._git_repository_key = None
        self._git_branch_name = None
        self._git_folder_path = None
        self._lifecycle_state = None

    @property
    def key(self):
        """
        Gets the key of this UpdateOntologyProjectDetails.

        :return: The key of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this UpdateOntologyProjectDetails.

        :param key: The key of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._key = key

    @property
    def display_name(self):
        """
        Gets the display_name of this UpdateOntologyProjectDetails.

        :return: The display_name of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this UpdateOntologyProjectDetails.

        :param display_name: The display_name of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._display_name = display_name

    @property
    def description(self):
        """
        Gets the description of this UpdateOntologyProjectDetails.

        :return: The description of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this UpdateOntologyProjectDetails.

        :param description: The description of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._description = description

    @property
    def namespace(self):
        """
        Gets the namespace of this UpdateOntologyProjectDetails.

        :return: The namespace of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._namespace

    @namespace.setter
    def namespace(self, namespace):
        """
        Sets the namespace of this UpdateOntologyProjectDetails.

        :param namespace: The namespace of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._namespace = namespace

    @property
    def workspace_base_path(self):
        """
        Gets the workspace_base_path of this UpdateOntologyProjectDetails.
        Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.


        :return: The workspace_base_path of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._workspace_base_path

    @workspace_base_path.setter
    def workspace_base_path(self, workspace_base_path):
        """
        Sets the workspace_base_path of this UpdateOntologyProjectDetails.
        Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.


        :param workspace_base_path: The workspace_base_path of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._workspace_base_path = workspace_base_path

    @property
    def source_type(self):
        """
        Gets the source_type of this UpdateOntologyProjectDetails.
        Project content source.

        Allowed values for this property are: "VOLUME", "GIT"


        :return: The source_type of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._source_type

    @source_type.setter
    def source_type(self, source_type):
        """
        Sets the source_type of this UpdateOntologyProjectDetails.
        Project content source.


        :param source_type: The source_type of this UpdateOntologyProjectDetails.
        :type: str
        """
        allowed_values = ["VOLUME", "GIT"]
        if not value_allowed_none_or_none_sentinel(source_type, allowed_values):
            raise ValueError(
                "Invalid value for `source_type`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._source_type = source_type

    @property
    def git_repository_key(self):
        """
        Gets the git_repository_key of this UpdateOntologyProjectDetails.
        Git repository key for git-backed ontology projects.


        :return: The git_repository_key of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._git_repository_key

    @git_repository_key.setter
    def git_repository_key(self, git_repository_key):
        """
        Sets the git_repository_key of this UpdateOntologyProjectDetails.
        Git repository key for git-backed ontology projects.


        :param git_repository_key: The git_repository_key of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._git_repository_key = git_repository_key

    @property
    def git_branch_name(self):
        """
        Gets the git_branch_name of this UpdateOntologyProjectDetails.
        Git branch name for git-backed ontology projects.


        :return: The git_branch_name of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._git_branch_name

    @git_branch_name.setter
    def git_branch_name(self, git_branch_name):
        """
        Sets the git_branch_name of this UpdateOntologyProjectDetails.
        Git branch name for git-backed ontology projects.


        :param git_branch_name: The git_branch_name of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._git_branch_name = git_branch_name

    @property
    def git_folder_path(self):
        """
        Gets the git_folder_path of this UpdateOntologyProjectDetails.
        Workspace-relative Git folder path for git-backed ontology project content.


        :return: The git_folder_path of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._git_folder_path

    @git_folder_path.setter
    def git_folder_path(self, git_folder_path):
        """
        Sets the git_folder_path of this UpdateOntologyProjectDetails.
        Workspace-relative Git folder path for git-backed ontology project content.


        :param git_folder_path: The git_folder_path of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._git_folder_path = git_folder_path

    @property
    def lifecycle_state(self):
        """
        Gets the lifecycle_state of this UpdateOntologyProjectDetails.

        :return: The lifecycle_state of this UpdateOntologyProjectDetails.
        :rtype: str
        """
        return self._lifecycle_state

    @lifecycle_state.setter
    def lifecycle_state(self, lifecycle_state):
        """
        Sets the lifecycle_state of this UpdateOntologyProjectDetails.

        :param lifecycle_state: The lifecycle_state of this UpdateOntologyProjectDetails.
        :type: str
        """
        self._lifecycle_state = lifecycle_state

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/UpdateOntologyProjectDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * 
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=UpdateOntologyProjectDetails.Builder.class)

public final class UpdateOntologyProjectDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"key", "displayName", "description", "namespace", "workspaceBasePath", "sourceType", "gitRepositoryKey", "gitBranchName", "gitFolderPath", "lifecycleState"})
    public UpdateOntologyProjectDetails(String key, String displayName, String description, String namespace, String workspaceBasePath, OntologyProjectSourceType sourceType, String gitRepositoryKey, String gitBranchName, String gitFolderPath, String lifecycleState) {
        super();
        this.key = key;
        this.displayName = displayName;
        this.description = description;
        this.namespace = namespace;
        this.workspaceBasePath = workspaceBasePath;
        this.sourceType = sourceType;
        this.gitRepositoryKey = gitRepositoryKey;
        this.gitBranchName = gitBranchName;
        this.gitFolderPath = gitFolderPath;
        this.lifecycleState = lifecycleState;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;



public Builder key(String key) {
    this.key = key;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;



public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;



public Builder description(String description) {
    this.description = description;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("namespace")
private String namespace;



public Builder namespace(String namespace) {
    this.namespace = namespace;
    return this;
}
            /**
     * Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("workspaceBasePath")
private String workspaceBasePath;

        /**
         * Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.
         * @param workspaceBasePath the value to set
         * @return this builder
         **/
        

public Builder workspaceBasePath(String workspaceBasePath) {
    this.workspaceBasePath = workspaceBasePath;
    return this;
}
            /**
     * Project content source.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("sourceType")
private OntologyProjectSourceType sourceType;

        /**
         * Project content source.
         * @param sourceType the value to set
         * @return this builder
         **/
        

public Builder sourceType(OntologyProjectSourceType sourceType) {
    this.sourceType = sourceType;
    return this;
}
            /**
     * Git repository key for git-backed ontology projects.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitRepositoryKey")
private String gitRepositoryKey;

        /**
         * Git repository key for git-backed ontology projects.
         * @param gitRepositoryKey the value to set
         * @return this builder
         **/
        

public Builder gitRepositoryKey(String gitRepositoryKey) {
    this.gitRepositoryKey = gitRepositoryKey;
    return this;
}
            /**
     * Git branch name for git-backed ontology projects.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitBranchName")
private String gitBranchName;

        /**
         * Git branch name for git-backed ontology projects.
         * @param gitBranchName the value to set
         * @return this builder
         **/
        

public Builder gitBranchName(String gitBranchName) {
    this.gitBranchName = gitBranchName;
    return this;
}
            /**
     * Workspace-relative Git folder path for git-backed ontology project content.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("gitFolderPath")
private String gitFolderPath;

        /**
         * Workspace-relative Git folder path for git-backed ontology project content.
         * @param gitFolderPath the value to set
         * @return this builder
         **/
        

public Builder gitFolderPath(String gitFolderPath) {
    this.gitFolderPath = gitFolderPath;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
private String lifecycleState;



public Builder lifecycleState(String lifecycleState) {
    this.lifecycleState = lifecycleState;
    return this;
}


        public UpdateOntologyProjectDetails build() {
            UpdateOntologyProjectDetails model = new UpdateOntologyProjectDetails(this.key
                , this.displayName
                , this.description
                , this.namespace
                , this.workspaceBasePath
                , this.sourceType
                , this.gitRepositoryKey
                , this.gitBranchName
                , this.gitFolderPath
                , this.lifecycleState);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(UpdateOntologyProjectDetails model) {
                this.key(model.getKey());
    this.displayName(model.getDisplayName());
    this.description(model.getDescription());
    this.namespace(model.getNamespace());
    this.workspaceBasePath(model.getWorkspaceBasePath());
    this.sourceType(model.getSourceType());
    this.gitRepositoryKey(model.getGitRepositoryKey());
    this.gitBranchName(model.getGitBranchName());
    this.gitFolderPath(model.getGitFolderPath());
    this.lifecycleState(model.getLifecycleState());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

    
    public String getKey() {
        return key;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

    
    public String getDisplayName() {
        return displayName;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

    
    public String getDescription() {
        return description;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("namespace")
    private final String namespace;

    
    public String getNamespace() {
        return namespace;
    }


        /**
     * Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("workspaceBasePath")
    private final String workspaceBasePath;

        /**
     * Root path for volume-backed ontology project content. Defaults to a workspace-relative path; managed-volume deployments may store this as an OMS managed-volume path.
     * @return the value
     **/
    
    public String getWorkspaceBasePath() {
        return workspaceBasePath;
    }

    
        /**
     * Project content source.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sourceType")
    private final OntologyProjectSourceType sourceType;

        /**
     * Project content source.
     * @return the value
     **/
    
    public OntologyProjectSourceType getSourceType() {
        return sourceType;
    }


        /**
     * Git repository key for git-backed ontology projects.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitRepositoryKey")
    private final String gitRepositoryKey;

        /**
     * Git repository key for git-backed ontology projects.
     * @return the value
     **/
    
    public String getGitRepositoryKey() {
        return gitRepositoryKey;
    }


        /**
     * Git branch name for git-backed ontology projects.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitBranchName")
    private final String gitBranchName;

        /**
     * Git branch name for git-backed ontology projects.
     * @return the value
     **/
    
    public String getGitBranchName() {
        return gitBranchName;
    }


        /**
     * Workspace-relative Git folder path for git-backed ontology project content.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("gitFolderPath")
    private final String gitFolderPath;

        /**
     * Workspace-relative Git folder path for git-backed ontology project content.
     * @return the value
     **/
    
    public String getGitFolderPath() {
        return gitFolderPath;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
    private final String lifecycleState;

    
    public String getLifecycleState() {
        return lifecycleState;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("UpdateOntologyProjectDetails(");
        sb.append("key=").append(String.valueOf(this.key));
        sb.append(", displayName=").append(String.valueOf(this.displayName));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", namespace=").append(String.valueOf(this.namespace));
        sb.append(", workspaceBasePath=").append(String.valueOf(this.workspaceBasePath));
        sb.append(", sourceType=").append(String.valueOf(this.sourceType));
        sb.append(", gitRepositoryKey=").append(String.valueOf(this.gitRepositoryKey));
        sb.append(", gitBranchName=").append(String.valueOf(this.gitBranchName));
        sb.append(", gitFolderPath=").append(String.valueOf(this.gitFolderPath));
        sb.append(", lifecycleState=").append(String.valueOf(this.lifecycleState));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof UpdateOntologyProjectDetails)) {
            return false;
        }

        UpdateOntologyProjectDetails other = (UpdateOntologyProjectDetails) o;
        return java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.namespace, other.namespace) &&
            java.util.Objects.equals(this.workspaceBasePath, other.workspaceBasePath) &&
            java.util.Objects.equals(this.sourceType, other.sourceType) &&
            java.util.Objects.equals(this.gitRepositoryKey, other.gitRepositoryKey) &&
            java.util.Objects.equals(this.gitBranchName, other.gitBranchName) &&
            java.util.Objects.equals(this.gitFolderPath, other.gitFolderPath) &&
            java.util.Objects.equals(this.lifecycleState, other.lifecycleState);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.namespace == null ? 43 : this.namespace.hashCode());
        result = (result * PRIME) + (this.workspaceBasePath == null ? 43 : this.workspaceBasePath.hashCode());
        result = (result * PRIME) + (this.sourceType == null ? 43 : this.sourceType.hashCode());
        result = (result * PRIME) + (this.gitRepositoryKey == null ? 43 : this.gitRepositoryKey.hashCode());
        result = (result * PRIME) + (this.gitBranchName == null ? 43 : this.gitBranchName.hashCode());
        result = (result * PRIME) + (this.gitFolderPath == null ? 43 : this.gitFolderPath.hashCode());
        result = (result * PRIME) + (this.lifecycleState == null ? 43 : this.lifecycleState.hashCode());
        return result;
    }


}
